tests/test_process.py

```diff
@@ -1106,6 +1106,142 @@ def test_process_signal_min_max(
     pd.testing.assert_series_equal(result, expected)
 
 
+@pytest.mark.parametrize(
+    'process_func, signal, sampling_rate',
+    [
+        (
+            lambda x, sr: x.mean(),
+            np.array([0, 0, 0, 0, 1, 1, 1, 1], dtype=np.float32),
+            1,
+        ),
+    ]
+)
+@pytest.mark.parametrize(
+    'start, end, win_dur, hop_dur, expected',
+    [
+        (
+            None, None, 4, None,
+            pd.Series(
+                [0, 0.5, 1],
+                audinterface.utils.signal_index(
+                    [0, 2, 4],
+                    [4, 6, 8],
+                ),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            None, None, 4, 2,
+            pd.Series(
+                [0, 0.5, 1],
+                audinterface.utils.signal_index(
+                    [0, 2, 4],
+                    [4, 6, 8],
+                ),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            1.0, None, 4, 2,
+            pd.Series(
+                [0.25, 0.75],
+                audinterface.utils.signal_index(
+                    [1, 3],
+                    [5, 7],
+                ),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            1.0, 5.0, 4, 2,
+            pd.Series(
+                [0.25],
+                audinterface.utils.signal_index(1, 5),
+                dtype=np.float32,
+            ),
+        ),
+        (
+            1.0, 2.0, 4, 2,
+            pd.Series(
+                [],
+                audinterface.utils.signal_index(),
+                dtype=object,
+            ),
+        ),
+        # missing win duration
+        pytest.param(
+            None, None, None, 2, None,
+            marks=pytest.mark.xfail(raises=ValueError),
+        )
+    ]
+)
+def test_process_with_sliding_window(
+        tmpdir,
+        process_func,
+        signal,
+        sampling_rate,
+        start,
+        end,
+        win_dur,
+        hop_dur,
+        expected,
+):
+    # save signal to file
+    root = tmpdir
+    file = 'file.wav'
+    path = os.path.join(root, file)
+    audiofile.write(path, signal, sampling_rate)
+
+    # create interface
+    process = audinterface.Process(
+        process_func=process_func,
+        hop_dur=hop_dur,
+        win_dur=win_dur,
+    )
+
+    # process signal
+    y = process.process_signal(
+        signal,
+        sampling_rate,
+        start=start,
+        end=end,
+    )
+    pd.testing.assert_series_equal(y, expected, atol=1.0e-4)
```

Comment:
Interesting, this first of the five tests works for me without `atol`. I'm also a little bit surprised that `numpy` is not able to end up with our easy numbers, but I guess this is how it is ;)

Should we maybe define the tolerance in a single variable in the parametrisation, where you specify `'process_func, signal, sampling_rate'`?

Reply:
The problem comes from rounding errors when storing the input signal to a WAV file. I now changed to 32-bit depth when storing the file, and now it works without `atol`.

Reply:
Ah, great.